### PR TITLE
Improve Device Synchronization and Tokenization Logic Across Components

### PR DESCRIPTION
**Pull request by uzair-mughal**
This pull request addresses several key updates and refinements:

- **Device Management:**
  - Updated `DeviceManager` to ensure device synchronization across all components that utilize device-specific operations.
  - Changed the default device selection to CPU instead of forcefully choosing the best available device.

- **Phoneme Tokenization:**
  - Updated `PhonemeTokenizer` to reposition the `[PAD]` token to the last dimension.
  - Replaced the hard-coded PAD index with a dynamic reference via `self.special_token_dims["[PAD]"]` during target setup.
  - Modified one-hot encoding logic to skip the PAD token using:
    ```python
    one_hot = torch.isin(torch.arange(self.vocabulary_size - 1, device=self.device), indices).long()
    ```

- **BridgeDataset Enhancements:**
  - Enabled batch encoding to improve data loading efficiency.
  - Temporarily disabled caching functionality until a robust batch-wise caching logic can be implemented.

- **Training Pipeline Adjustments:**
  - Updated loss calculations to properly ignore PAD tokens:
    ```python
    phon_loss = torch.nn.CrossEntropyLoss(ignore_index=35)(
                     logits["phon"], phonology["targets"]
                 )
    
    orth_loss = torch.nn.CrossEntropyLoss(ignore_index=2)(
                     logits["orth"], orthography["enc_input_ids"][:, 1:]
                 )
    ```
    - These modifications ensure that the PAD token is correctly ignored during the loss computation.

These changes aim to enhance the stability and flexibility of the system, ensuring a more robust training pipeline and improved handling of special tokens.

*Note: I apologize for the extra changes in other files caused by my formatter.*

### CODE REVIEW

**Review by mathnathan — APPROVED** [review]
Excellent work! Thank you for this Herculean effort
